worlds/kh1/Rules.py

```diff
@@ -0,0 +1,2142 @@
+from BaseClasses import CollectionState, MultiWorld
+
+
+def has_x_worlds(state: CollectionState, player: int, num_of_worlds: int, keyblades_unlock_chests: bool) -> bool:
+    worlds_acquired = 0.0
+    worlds =    ["Wonderland", "Olympus Coliseum", "Deep Jungle", "Agrabah",      "Monstro",      "Atlantica", "Halloween Town", "Neverland",  "Hollow Bastion", "End of the World"]
+    keyblades = ["Lady Luck",  "Olympia",          "Jungle King", "Three Wishes", "Wishing Star", "Crabclaw",  "Pumpkinhead",    "Fairy Harp", "Divine Rose",    "Oblivion"]
+    for i in range(len(worlds)):
+        if state.has(worlds[i], player):
+            worlds_acquired = worlds_acquired + 0.5
+        if (state.has(worlds[i], player) and has_keyblade(state, player, keyblades_unlock_chests, keyblades[i])) or (state.has(worlds[i], player) and worlds[i] == "Atlantica"):
+            worlds_acquired = worlds_acquired + 0.5
+    return worlds_acquired >= num_of_worlds
+
+def has_slides(state: CollectionState, player: int) -> bool:
+    return state.has("Slides", player)
+
+def has_evidence(state: CollectionState, player: int) -> bool:
+    return state.has("Footprints", player) #or state.has("Stench", player) or state.has("Claw Marks", player) or state.has("Antenna", player)
+
+def can_glide(state: CollectionState, player: int) -> bool:
+    return state.has("Progressive Glide", player) or state.has("Superglide", player)
+
+def has_emblems(state: CollectionState, player: int, keyblades_unlock_chests: bool) -> bool:
+    return (
+                state.has("Emblem Piece (Flame)", player)
+                and state.has("Emblem Piece (Chest)", player)
+                and state.has("Emblem Piece (Statue)", player)
+                and state.has("Emblem Piece (Fountain)", player)
+                and state.has("Hollow Bastion", player)
+                and has_x_worlds(state, player, 5, keyblades_unlock_chests)
+           )
+
+def has_item(state: CollectionState, player: int, item: str) -> bool:
+    return state.has(item, player)
+
+def has_at_least(state: CollectionState, player: int, item: str, x: int) -> bool:
+    return state.count(item, player) >= x
```

Comment:
```suggestion
def has_at_least(state: CollectionState, player: int, item: str, x: int) -> bool:
    return state.has(item, player, x)
```
I don't really see the point of this and the previous function, but it's a bit late to change it so at least you can change this to use the quantity field for state.has